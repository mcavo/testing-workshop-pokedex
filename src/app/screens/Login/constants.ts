import { AuthData } from '@interfaces/authInterfaces';

export const FIELDS = {
  email: 'email',
  password: 'password'
};

export const LOGIN_INITIAL_VALUES: AuthData = {
  email: '',
  password: ''
};
