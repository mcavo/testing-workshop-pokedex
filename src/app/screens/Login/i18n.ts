import i18next from 'i18next';

i18next.addResources('en', 'LOGIN', {
  POKEDEX: 'Pokedex',
  SUBMIT: 'Log in'
});

i18next.addResources('es', 'LOGIN', {
  POKEDEX: 'Pokedex',
  SUBMIT: 'Ingresar'
});
