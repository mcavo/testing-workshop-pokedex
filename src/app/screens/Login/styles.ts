import { StyleSheet } from 'react-native';
import { lightGray } from '@constants/colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: lightGray,
    paddingHorizontal: 20,
    paddingVertical: 10
  },
  title: {
    marginVertical: 30
  }
});
