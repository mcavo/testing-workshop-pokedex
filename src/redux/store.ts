import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import Reactotron from 'reactotron-react-native';
import thunk from 'redux-thunk';
import { fetchMiddleware, configureMergeState } from 'redux-recompose';
import { ImmutableObject } from 'seamless-immutable';
import { IState } from '@interfaces/reduxInterfaces';

import auth from './auth/reducer';
import pokemon from './pokemon/reducer';

configureMergeState((state: ImmutableObject<IState>, diff: IState) => state.merge(diff));

const reducers = combineReducers({
  auth,
  pokemon
});

const middlewares = [];
const enhancers = [];

/* ------------- Thunk Middleware ------------- */
middlewares.push(thunk);

/* ------------- Redux-Recompose Middleware ------------- */
middlewares.push(fetchMiddleware);

/* ------------- Assemble Middleware ------------- */
enhancers.push(applyMiddleware(...middlewares));

if (__DEV__ && Reactotron.createEnhancer) enhancers.push(Reactotron.createEnhancer(true));

// In DEV mode, we'll create the store through Reactotron
const store = createStore(reducers, compose(...enhancers));

if (__DEV__ && Reactotron.setReduxStore) Reactotron.setReduxStore(store);

export default store;
