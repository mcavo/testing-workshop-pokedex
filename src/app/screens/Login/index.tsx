import React from 'react';
import { View, SafeAreaView } from 'react-native';
import { useForm } from 'react-hook-form';
import { useDispatch } from 'react-redux';
import i18next from 'i18next';
import CustomButton from '@components/CustomButton';
import CustomController from '@components/CustomController';
import CustomText from '@components/CustomText';
import { AuthData } from '@interfaces/authInterfaces';
import AuthActions from '@redux/auth/actions';
import { emailRules, passwordRules } from '@utils/validationsUtils';

import './i18n';

import styles from './styles';

function Login() {
  const dispatch = useDispatch();
  const { control, handleSubmit, errors } = useForm<AuthData>();
  const onSubmit = (values: AuthData) => {
    dispatch(AuthActions.login(values));
  };
  return (
    <View style={styles.container}>
      <SafeAreaView>
        <CustomText gray xbig center semiBold style={styles.title}>
          {i18next.t('LOGIN:POKEDEX')}
        </CustomText>
        <CustomController
          control={control}
          name="email"
          rules={emailRules}
          inputProps={{
            placeholder: i18next.t('FIELDS:EMAIL'),
            error: errors?.email?.message
          }}
        />
        <CustomController
          control={control}
          name="password"
          rules={passwordRules}
          inputProps={{
            placeholder: i18next.t('FIELDS:PASSWORD'),
            error: errors?.password?.message
          }}
        />
        <CustomButton text={i18next.t('LOGIN:SUBMIT')} onPress={handleSubmit(onSubmit)} small />
      </SafeAreaView>
    </View>
  );
}

export default Login;
