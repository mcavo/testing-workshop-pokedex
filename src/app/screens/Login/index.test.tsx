import React from 'react';
import { render } from '@testsUtils/redux';

import Login from './index';

describe('<Login />', () => {
  test('Login Snapshot', () => {
    const login = render(<Login />).toJSON();
    expect(login).toMatchSnapshot();
  });
});
