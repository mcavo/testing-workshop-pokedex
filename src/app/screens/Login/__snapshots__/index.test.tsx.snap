// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Login Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E5E5E5",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#555555",
            "fontFamily": "Montserrat",
            "fontSize": 25.142857142857142,
            "fontStyle": "normal",
            "fontWeight": "400",
          },
          Array [
            Object {
              "color": "#555555",
              "fontFamily": "Montserrat",
              "fontSize": 25.142857142857142,
              "fontStyle": "normal",
              "fontWeight": "600",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#A7A7A7",
            },
            Object {
              "fontSize": 56.57142857142857,
            },
          ],
          Object {
            "marginVertical": 30,
          },
        ]
      }
    >
      LOGIN:POKEDEX
    </Text>
    <View>
      <TextInput
        allowFontScaling={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="FIELDS:EMAIL"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#E5E5E5",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 10,
            },
            Object {},
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#555555",
              "fontFamily": "Montserrat",
              "fontSize": 25.142857142857142,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Array [
              Object {
                "fontSize": 18.857142857142858,
              },
              Object {
                "color": "#F04338",
              },
            ],
            Object {
              "marginBottom": 14,
              "marginLeft": 10,
              "marginTop": 6,
              "maxHeight": 22,
            },
          ]
        }
      />
    </View>
    <View>
      <TextInput
        allowFontScaling={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="FIELDS:PASSWORD"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#E5E5E5",
              "borderRadius": 4,
              "borderWidth": 1,
              "padding": 10,
            },
            Object {},
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#555555",
              "fontFamily": "Montserrat",
              "fontSize": 25.142857142857142,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Array [
              Object {
                "fontSize": 18.857142857142858,
              },
              Object {
                "color": "#F04338",
              },
            ],
            Object {
              "marginBottom": 14,
              "marginLeft": 10,
              "marginTop": 6,
              "maxHeight": 22,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 85.71428571428571,
            "justifyContent": "center",
            "minWidth": 300,
            "paddingHorizontal": 64.28571428571428,
          },
          Object {
            "backgroundColor": "#288CCB",
            "borderColor": "#FFF",
            "borderWidth": 0,
            "shadowColor": "#F1F1F1",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 10,
            "width": "auto",
          },
          Array [
            Object {
              "minWidth": 300,
            },
          ],
          undefined,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#555555",
              "fontFamily": "Montserrat",
              "fontSize": 25.142857142857142,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Array [
              Object {
                "fontSize": 22,
              },
              Object {
                "fontFamily": "Montserrat",
                "fontStyle": "normal",
                "fontWeight": "600",
              },
            ],
            Array [
              Object {
                "color": "#FFF",
              },
              Array [
                undefined,
              ],
              undefined,
            ],
          ]
        }
      >
        LOGIN:SUBMIT
      </Text>
    </View>
  </RCTSafeAreaView>
</View>
`;
